src/config/sqs.ts

```diff
@@ -0,0 +1,105 @@
+import AWS, {AWSError} from "aws-sdk";
+import Logger from "bunyan"
+import {getLogger} from "./logger"
+import {Message, ReceiveMessageResult} from "aws-sdk/clients/sqs";
+import envVars from "./env";
+
+// All our queues must be in the same region so we can set a global parameter here
+AWS.config.update({region: envVars.SQS_WEBHOOKSQS_QUEUE_REGION || "us-west-1"});
+
+// Create SQS service client
+const sqs = new AWS.SQS({apiVersion: "2012-11-05"});
+const logger = getLogger("config.sqs")
+
+// Replace with your account id and the queue name you setup
+
+export class SqsQueue {
+	queueUrl: string;
+	queueName: string;
+
+	constructor(queueName: string, queueUrl: string) {
+		this.queueUrl = queueUrl;
+		this.queueName = queueName;
+	}
+
+	sendMessage(payload: any, log?: Logger) {
+		log = log || logger
+		const params = {
+			MessageBody: JSON.stringify(payload),
+			QueueUrl: this.queueUrl
+		};
+		sqs.sendMessage(params, (err, data) => {
+			if (err) {
+				log.error(err, "Error sending SQS message");
+			} else {
+				log.info(`Successfully added message to sqs queue messageId: ${data.MessageId}`);
+			}
+		});
+	}
+
+	deleteMessage(message: Message) {
+		const deleteParams = {
+			QueueUrl: this.queueUrl,
+			ReceiptHandle: message.ReceiptHandle
+		};
+		sqs.deleteMessage(deleteParams, (err, data) => {
+			if (err) {
+				logger.error({err, data}, "Error deleting message from the queue");
+			} else {
+				logger.debug({data}, "Successfully deleted message from queue");
+			}
+		});
+	}
+
+	executeMessage(message: Message, messageHandler) {
+		const messagePayload = JSON.parse(message.Body);
+		logger.info({messagePayload}, "Sqs message received");
+
+		try {
+			messageHandler({data: messagePayload})
+			this.deleteMessage(message)
+		} catch (err) {
```

Comment:
We need to throw a specific error when hitting a rate limit and then catch it here and re-send the message to SQS with a delay from the rate limit header.